compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/DecomposeWinogradPass.cpp

```diff
@@ -82,6 +82,94 @@ struct FoldWinogradOpUnitDims : public OpRewritePattern<TransformOp> {
+/// ````
+struct DecomposeWinogradFilterTransform
+    : public OpRewritePattern<WinogradFilterTransformOp> {
+  using OpRewritePattern<WinogradFilterTransformOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(WinogradFilterTransformOp transformOp,
+                                PatternRewriter &rewriter) const override {
+    Location loc = transformOp.getLoc();
+    auto funcOp = transformOp->getParentOfType<mlir::FunctionOpInterface>();
+    if (!funcOp) {
+      return rewriter.notifyMatchFailure(transformOp,
+                                         "Could not find parent function");
+    }
+    rewriter.setInsertionPointToStart(&funcOp.getFunctionBody().front());
+
+    Value inputSlice = transformOp.input();
+    Value outputSlice = transformOp.output();
+    if (transformOp.getInputRank() != 2 || transformOp.getOutputRank() != 2) {
+      return rewriter.notifyMatchFailure(transformOp, "Winograd op not tiled");
+    }
+    const int64_t inputTileSize = transformOp.getInputTileSize();
+    const int64_t kernelSize = transformOp.getKernelSize();
+    ArrayRef<int64_t> kernelDims = transformOp.getKernelDimensions();
+    llvm::SmallSetVector<int64_t, 2> kernelDimsSet(kernelDims.begin(),
+                                                   kernelDims.end());
+    Type elementType = transformOp.getOutputType().getElementType();
+    Value zeroF32 = rewriter.create<arith::ConstantOp>(
+        loc, rewriter.getZeroAttr(elementType));
+    const float *GT{nullptr};
+    const float *G{nullptr};
+    GT = IREE::LinalgExt::Winograd::GT_6x6_3x3;
+    G = IREE::LinalgExt::Winograd::G_6x6_3x3;
+    Value GTV = IREE::LinalgExt::createValueFrom2DConstant(
+        GT, kernelSize, inputTileSize, loc, rewriter);
+    Value GV = IREE::LinalgExt::createValueFrom2DConstant(
+        G, inputTileSize, kernelSize, loc, rewriter);
+
+    // Create computation
+    OpBuilder::InsertionGuard afterTransformOp(rewriter);
+    rewriter.setInsertionPointAfter(transformOp);
+
+    // Create matmul(input, GT)
+    SmallVector<int64_t> initShape(kernelDims.size(), inputTileSize);
+    initShape[0] = kernelSize;
+    auto tensorType = cast<ShapedType>(outputSlice.getType()).clone(initShape);
+    Value init = rewriter.create<tensor::EmptyOp>(loc, initShape, elementType);
+    linalg::FillOp fillOp = rewriter.create<linalg::FillOp>(
+        loc, ValueRange{zeroF32}, ValueRange{init});
+    linalg::MatmulOp matmulOp = rewriter.create<linalg::MatmulOp>(
+        loc, tensorType, ValueRange{inputSlice, GTV}, fillOp.result());
+
+    // Create matmul(G, matmul(input, GT))
+    fillOp = rewriter.create<linalg::FillOp>(loc, ValueRange{zeroF32},
+                                             ValueRange{outputSlice});
+    matmulOp = rewriter.create<linalg::MatmulOp>(
+        loc, outputSlice.getType(), ValueRange{GV, matmulOp.getResult(0)},
+        fillOp.result());
+    transformOp.getResult()[0].replaceAllUsesWith(matmulOp.getResult(0));
```

Comment:
Why not use `rewriter.replaceOp()` method? It is much more common in pattern-rewriting way.